source/common/upstream/cluster_factory_impl.cc

```diff
@@ -57,6 +57,13 @@ std::pair<ClusterSharedPtr, ThreadAwareLoadBalancerPtr> ClusterFactoryImplBase::
   } else {
     cluster_type = cluster.cluster_type().name();
   }
+
+  if (cluster.common_lb_config().use_hostname() &&
+      !(cluster.type() == envoy::config::cluster::v3::Cluster::STRICT_DNS ||
+        cluster.type() == envoy::config::cluster::v3::Cluster::LOGICAL_DNS)) {
```

Comment:
Does this even make sense for logical_dns? I can't see anyone using logical DNS with a hashing LB as there is only ever a single logical host? Perhaps this should only be an option for strict DNS and we should actually have a specific strict DNS configuration of some type vs. having the config be global to all lb options? WDYT?

Reply:
My understanding might be a bit flawed but here's what I'm thinking:

This makes the most sense for StrictDNS which returns just a single IP address. It doesn't quite make sense for StrictDNS with multiple IP addresses because the Cluster will have multiple hosts but the same hostname. So the ring/maglev table will have multiple entries with the same hash but different IP. 

For LogicalDNS with single IP this works well as expected although should probably use StrictDNS. For LogicalDNS with multiple/changing IPs because there is only one host in envoy this can still work if the IP is rotated for failover reasons but not if it is rotated for loadbalancing. Still seems like it doesn't make sense here.

If we restrict it to just StrictDNS, where would you want the config to be? There isn't something like this afaik. I still think common_lb_config makes the most sense because it is common to the hashing loadbalancers and this config is passed into the constructors which simplifies the implementation.

Reply:
Sure, that makes sense. But if we are going to add another message then what about doing what I did in the previous attempt? Add a `MaglevLbConfig` to compliment the `RingHashLbConfig` and both have the `use_hostname_for_consistent_hashing` as options within them.

I'm good with either solution. Let me know what you prefer and I'll make it happen.

Reply:
> I'm good with either solution. Let me know what you prefer and I'll make it happen.

I could go either way on this. In some sense a common config for all hashing load balancers is more future proof so that would be my mild preference?

Reply:
One option would be to add the MaglevLbConfig, then also add a `CommonConsistentHashingLbConfig` and include that from both configs? Up to you.

Reply:
Sorry where did we land here about supporting this for LogicalDNS? I'm still not convinced this makes sense? Should we only support this for StrictDNS where the user supplies  a set of host names that they want to route to?